src/RuleSet/Sets/PhpCsFixerSet.php

```diff
@@ -110,6 +110,32 @@ public function getRules(): array
             'php_unit_test_class_requires_covers' => true,
             'phpdoc_add_missing_param_annotation' => true,
             'phpdoc_no_empty_return' => true,
+            'phpdoc_order' => [
+                'order' => [
+                    'deprecated',
+                    'internal',
+                    'final',
+                    'readonly',
+                    'immutable',
+                    'type',
+                    'template',
+                    'template-covariant',
+                    'template-extends',
+                    'extends',
+                    'implements',
+                    'property',
+                    'method',
+                    'param',
+                    'return',
+                    'var',
+                    'assert',
+                    'assert-if-false',
```

Comment:
Sorry, but I have to get back to the topic of these tags and their order. This config is (mostly and in the first place) for this repo itself, but out of 22 tags set here, in the repo only 13 are present, and those 13 should(`*`) be in the order:
```php
['deprecated', 'final', 'extends', 'implements', 'return', 'template', 'internal', 'method', 'author', 'throws', 'param', 'var', 'see'];
```

`*` - according to the order in which they are most often used in the project at the moment

Reply:
I put a lot of effording in analysing "mostly used" and "mostly logical" order.

Might I know how did you come up with these?

Let me share a few "definitely right order pairs":
- `@template` must come before `@implements` as the later might want to refer to the first
- `@author` is unrelated to "meaningful tags", so it should come after these, simillar for `@see`
- `@return` in definitely more logical to come after `@param`
- `@internal`, `@deprecated` as semantically related, so they should be placed "more or less together"

Reply:
I agree with both of you:

- there's no need to define tags that are not used across the project. If new tags are introduced, ruleset can be modified accordingly. But at the same time I am _not_ strongly against including additional tags, if these are likely to be used at some point.
- I wouldn't follow current order as it was more or less random, I like the reasoning provided by @mvorisek and I think tags should be ordered by importance and grouped semantically.

Reply:
All commonly used tags must be present as the `PhpCsFixerSet` ruleset is exported/public and we do not know what tags are used by other projects. Also - related with https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8220#discussion_r1799147064 - the current order in `PhpCsFixerSet` contains exactly all tags ordered by default, but only with improved `@throws` order (which is planned for v4, but for BC, the default order is kept).

Reply:
- local config file for aligning on this repo-only configuration
- `PhpCsFixerSet` is the recommendation we are giving to developers for their projects. Changing it is sensitive (but for sure possible) - yet, I would like to validate if any established community has their recommendation already, before we introduce our new one.

Reply:
> All commonly used tags must be present as the PhpCsFixerSet ruleset is exported/public and we do not know what tags are used by other projects.

The thing is, we don't care. It's our own ruleset, and it should be focused around our own coding standards and phpDoc tag usage. If anyone is using `@PhpCsFixer` as a base for their own ruleset, then it's up to them to override this particular fixer's config to fit their needs. We shouldn't configure our internal ruleset with userland in mind (also this is why changes in the rulesets are _not_ covered with BC promise, as these are a living standards, in contrast to standalone rules' default configuration).

Reply:
> Might I know how did you come up with these?

By this: https://gist.github.com/kubawerlos/cdd4cf7aa4b2ecaaea88c62bb3022aee - I've calculated how each pair of tags currently occurs.

Reply:
> It's our own ruleset, and it should be focused around our own coding standards (...)

actually, it's ruleset I always encourage everyone to use. Opinionated, but I still encourage this one if they can afford sth more hardcore than PSR/PER-CS.

Reply:
We can encourage, but it does not change what I wrote - it's a living standard everyone agrees on when opting-in so CS fixes after updating Fixer are not surprising. It's different from CS changes coming from the single rule that someone has included in custom config (not based on ruleset).

Reply:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8220#discussion_r1799652691 with data shown in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8220#discussion_r1798461558 seems somehow odd to me, as:
- I do not belive `@return` comes before `@param` here or any coding standard
- and it is complete as `@phpstan-assert-if-true` was added by #8236,
- even if it would be complete, `@phpstan-assert-if-false` is complementary and should be added therefore.

The order is well reasoned in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8220#discussion_r1798484872.

Let's merge this PR as is.